GiniVision/Classes/GiniPDFDocument.swift

```diff
@@ -56,6 +57,42 @@ final public class GiniPDFDocument: NSObject, GiniVisionDocument {
         }
     }
     
+    fileprivate func renderFirstPage(fromPdf pdf:CGPDFDocument) -> UIImage? {
+        var pdfImage:UIImage?
+        let pdfDoc = pdf
+        
+        if let pdfPage:CGPDFPage = pdfDoc.page(at: 1) {
+            var pageRect:CGRect = pdfPage.getBoxRect(.cropBox)
+            
+            // In case that the image was rotated 90 or 270, final rect should be rotated to portrait             
+            if pdfPage.rotationAngle == 90 || pdfPage.rotationAngle == 270 {
+                let tempWidth = pageRect.size.width
+                pageRect.size.width = pageRect.size.height
+                pageRect.size.height = tempWidth
+            }
+            
+            // Create context
+            UIGraphicsBeginImageContextWithOptions(CGSize(width: pageRect.width, height: pageRect.height), false, 0.0)
+            let context:CGContext = UIGraphicsGetCurrentContext()!
+            
+            // Fill context color
+            context.setFillColor(UIColor.white.cgColor)
+            context.fill(pageRect)
+            
+            // Align PDF's cropBox to the context
```

Comment:
This one would be the only comment here that would be useful. However, it's vague and I still don't understand it.